engine/src/flutter/shell/platform/linux/fl_renderer.h

```diff
@@ -55,6 +55,14 @@ struct _FlRendererClass {
    * @renderer: an #FlRenderer.
    */
   void (*clear_current)(FlRenderer* renderer);
+
+  /**
+   * Virtual method responsible for scheduling the callback on the main thread.
+   * Overriden in the mock renderer.
+   */
+  gboolean (*schedule_on_main_thread)(FlRenderer* renderer,
+                                      void (*callback)(gpointer data),
+                                      gpointer data);
```

Comment:
This is a bit of a red-flag, why does this need to be a virtual method? Why can't we get the tests to work without overriding this?

Reply:
This is now gone.